build.cake

```diff
@@ -71,10 +71,10 @@ Task("Test")
 
 Task("UploadTestReport")
     .IsDependentOn("Test")
-    .WithCriteria((context) => FileExists(testCoverageOutputFilePath))
-    .WithCriteria(!string.IsNullOrWhiteSpace(coverallsRepoToken))
-    .WithCriteria((context) => !BuildSystem.IsPullRequest)
-    .WithCriteria((context) => !BuildSystem.IsLocalBuild)
+    .WithCriteria(context => FileExists(testCoverageOutputFilePath) &&
```

Comment:
Could you please explain what is the benefit of these changes?
How does it make better readability?